tests/fs-test.rs

```diff
@@ -11,13 +11,13 @@ fn new_file_test() {
 	let testfile = PathBuf::from("foo.txt");
 	if testfile.exists() {
 		println!("Removing existing file {}", testfile.display());
-		remove_file(&testfile).expect(&std::format!("Can't remove {}", testfile.display()));
```

Comment:
What exactly is the problem with the expects?

Reply:
[`clippy::expect_fun_call`](https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call).

This `format!` eagerly allocates a string, even when the result is ok. We could rewrite that to keep the semantics, but that would be unnecessarily unwieldy in this case, since three lines above you can see that `testfile` is just `foo.txt`.

Reply:
Then I'd prefer the suggested fix from the clippy page:
```rust
foo.unwrap_or_else(|| panic!("Err {}: {}", err_code, err_msg));
```

Reply:
Done!